File: DSA/Aanidhay Aggarwal/Day2_q1.java
import java.util.Scanner;

public class Main {
    public static String classCancelled() {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the threshold number of students: ");
        int threshold = scanner.nextInt();

        System.out.print("Enter the number of students: ");
        int n = scanner.nextInt();

        int[] arrivalTimes = new int[n];
        System.out.print("Enter the arrival times (space-separated): ");
        for (int i = 0; i < n; i++) {
            arrivalTimes[i] = scanner.nextInt();
        }

        int onTimeCount = 0;
        for (int arrivalTime : arrivalTimes) {
            if (arrivalTime <= 0) {
                onTimeCount++;
            }
        }

        return (onTimeCount < threshold) ? "YES" : "NO";
    }

    public static void main(String[] args) {
        System.out.println(classCancelled());
    }
}

File: DSA/Hritvik_Bhandari/DAY2/day2q1.cpp
#include <iostream>
using namespace std;
bool classav(int arr[],int size,int threshold){
    int count=0;
    for(int i = 0 ; i<size;i++){
        if(arr[i]<0){
             count++;
        }
    }
    if(count<threshold){
        return false;
    }
    return true;
}
int main()
{
    int students;
    cout<<"Enter the number of students ";
    cin>>students;
    int threshold;
    cout<<"enter the threshold no. of students ";
    cin>>threshold;
    int arrival_time[students];
    for(int i = 0;i<students;i++){
        cin>>arrival_time[i];
    }
   int lecture = classav(arrival_time,students,threshold);
   if(lecture) 
      cout<<"NO";
   else 
      cout<<"YES";
   return 0;
}
